controllers/projectController.js

```diff
@@ -101,7 +104,47 @@ const controller = {
     return res.json({ project, createdBy, comments, jobs });
   },
   followProject: async (req, res) => {
-    res.send('hello');
+    try {
+      const user = await UserModel.findOne({ username: req.params.username }, { _id: 1 }).lean();
+      const project = await ProjectModel.findOne({ slug: req.params.slug }, { _id: 1 }).lean();
+      // Check if user has already followed project and create relationship if not
+      const userAlreadyFollowing = await ProjectsRelationshipModel.findOne(
+        { user_id: user._id, project_id: project._id },
```

Comment:
do you want to try `findOneAndUpdate` method with `upsert: true` ?
basically it looks for a document with the filters. if no such document exists, it will create one. could be a bit shorter than this

Reply:
Thanks for the tip. Have updated!